src/Arcus.Observability.Telemetry.Core/Logging/RequestLogEntry.cs

```diff
@@ -64,24 +70,74 @@ public class RequestLogEntry
             int statusCode,
             TimeSpan duration,
             IDictionary<string, object> context)
+            : this(method, host, uri, operationName, statusCode, sourceSystem: RequestSourceSystem.Http, duration, requestTime: DateTimeOffset.UtcNow.ToString(FormatSpecifiers.InvariantTimestampFormat), context)
         {
-            Guard.For<ArgumentException>(() => host?.Contains(" ") == true, "Requires a HTTP request host name without whitespace");
+            Guard.For<ArgumentException>(() => host?.Contains(" ") is true, "Requires a HTTP request host name without whitespace");
             Guard.NotNullOrWhitespace(operationName, nameof(operationName), "Requires an operation name that is not null or whitespace");
             Guard.NotLessThan(statusCode, 100, nameof(statusCode), "Requires a HTTP response status code that's within the 100-599 range to track a HTTP request");
             Guard.NotGreaterThan(statusCode, 599, nameof(statusCode), "Requires a HTTP response status code that's within the 100-599 range to track a HTTP request");
             Guard.NotLessThan(duration, TimeSpan.Zero, nameof(duration), "Requires a positive time duration of the request operation");
-            
+        }
+
+        private RequestLogEntry(
+            string method, 
+            string host, 
+            string uri, 
+            string operationName,
+            int statusCode,
+            RequestSourceSystem sourceSystem,
+            TimeSpan duration,
+            string requestTime,
```

Comment:
Hihi, we already do that. You see, this ctor is `private` and is called from `static` methods where we do the conversion from `DateTimeOffset` to `string` so we get a consistent `string` representation of this. We do need it as a `string` bc these types will be used as serialization, and with this, we can control how the `DateTimeOffset` is serialized (see `FormatSpecifiers.InvariantTimestampFormat`).

Reply:
I just thought it would be best to do that as late as possible, and thus, pass it as a datetimeoffset to the private ctor, and do the 'ToString' stuff inside the ctor.

Reply:
Hihi, yeah, I think this is this kind of style where you put no functionality in ctors and only assignments. I think we did that around Arcus that way, but we can check.

Reply:
I'm OK with it like it is for now.   It's not that it will be a breaking change should we ever refactor this.
IMHO, it's just best do to this as late as possible.